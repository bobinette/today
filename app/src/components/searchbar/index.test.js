import React from 'react';
import { shallow, mount } from 'enzyme';

import SearchBar from '.';

test('render SearchBar', () => {
  const props = { q: '', onChange: jest.fn() };

  const tree = shallow(<SearchBar {...props} />);
  expect(tree).toMatchSnapshot();
});

test('test the onChange is called', () => {
  const props = { q: '', onChange: jest.fn() };

  const wrapper = mount(<SearchBar {...props} />);
  wrapper
    .find('.SearchBar__input')
    .simulate('change', { target: { value: 'value' } });
  expect(props.onChange).toHaveBeenCalledWith('value');
});
