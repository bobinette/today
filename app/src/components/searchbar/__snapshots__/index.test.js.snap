// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render SearchBar 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    className=""
    onChange={[MockFunction]}
    q=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <i
          className="fas fa-search"
        />,
        <input
          className="Input SearchBar__input"
          onChange={[Function]}
          placeholder="Search your logs..."
          value=""
        />,
      ],
      "className": "SearchBar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fas fa-search",
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "Input SearchBar__input",
          "onChange": [Function],
          "placeholder": "Search your logs...",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <i
            className="fas fa-search"
          />,
          <input
            className="Input SearchBar__input"
            onChange={[Function]}
            placeholder="Search your logs..."
            value=""
          />,
        ],
        "className": "SearchBar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fas fa-search",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "Input SearchBar__input",
            "onChange": [Function],
            "placeholder": "Search your logs...",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
